### DISCUSSION TITLE
Implement a way to send from an alias / automatically create a reverse alias by following a pattern without the need to manually create a reverse alias using the dashboard

### DISCUSSION THREAD

**Discussion by Trainax** [simple-login/app#1770] [Feature Requests]
Hello everyone. When initiating a new conversation from an alias it is necessary to manually create a reverse alias for every contact we want to write an email to.

Let's say I want to write an email to `bob@example.com` from my alias `friends@alias.mydomain.com`. To do this I need to:

1. Open the Simplelogin dashboard (I don't like using the browser extension)
2. Click on the `Contacts` button of the alias I want to write an email from
3. Input the email address I want to send an email to
4. Create a reverse alias for that contact (the reverse alias will be in the `bob_at_example_com_somethingRandom@simplelogin.co` format)
5. Copy the reverse alias in my email client
6. Write and send the email

What I propose is to link a single `somethingRandom` string to each alias so that to write a new email what I would need to do is to simply follow the pattern to create a reverse alias directly from my email client.

Let me make an example: let's say that for my alias `friends@alias.mydomain.com` the random string is `aaabbb`.

- If I want to write an email to `alice@example.com` all I would need to do is to write an email to `alice_at_example_com_aaabbb@simplelogin.co` without the need to manually create a reverse alias in the dashboard, because the `aaabbb` string is uniquely tied to my alias and every email following the correct pattern which is sent from an authorized mailbox to that reverse alias would automatically add that reverse alias in the contacts list and deliver the email
- Similarly, if I want to write an email to `jack@example.com` all I would need to do is to write an email to `jack_at_example_com_aaabbb@simplelogin.co` from an authorized mailbox

This feature would of course be optional so that if someone wants to have a random string for each contact in each alias it would remain possibile.

Let me know what you think!

**Comment by crazy-matt** [comment]
This could potentially be implemented with the VERP technique similarly to what is described in that blog [article](https://addy.io/blog/sending-email-from-an-alias-and-updated-replies/) which can enable one to send an email even before creating an alias contact if the right `To:` field value is being used.
In term of use, the pattern might be seen a bit complex to memorise but as soon as you get used to it, it's super slick. 

You want to send an email to `hello@example.com` from `first@my-sub.aleeas.com`, you enter in the `To:` field:
```
To: <first+hello=example.com@my-sub.aleeas.com>
```

If the feature is implemented, one would rather use that one single step approach to create the alias and reverse alias on the fly instead of having to use the SL dashboard.

This issue, I believe, has also been reported [here](https://github.com/simple-login/app/discussions/1892).
Would love to see it gaining more attraction.

**Reply by Collabeta** [reply]
The lack of this is probably going to force me to find a different solution when I already pretty much settled on simple login being what I needed for a number of reasons.

It's just not practical to have my family members with aliases linked to our custom domain ask me to create a reverse alias whenever they have to email someone new!

**Reply by nguyenkims** [reply]
There's a pull request to add support for SMTP but unfortunately the author doesn't seem active anymore.
You can transfer an alias to your family member so they can easily create reverse aliases if they want.

**Reply by Collabeta** [reply]
Thanks for pointing that out. I hadn't realised it was possible to do a workaround using a free account and transfer.

I still hope an SMTP option gets picked up as this simplifies things for average users somewhat, especially where emailing new contacts tends to be infrequent and maybe be overlooked.

**Reply by trikotret** [reply]
> Thanks for pointing that out. I hadn't realised it was possible to do a workaround using a free account and transfer.
> 
> I still hope an SMTP option gets picked up as this simplifies things for average users somewhat, especially where emailing new contacts tends to be infrequent and maybe be overlooked.

I use SMTP2GO to initiate an email. It works great until SL decides to add their own but I'm not holding my breath.
